lib/ansible/modules/cloud/amazon/iam_role.py

```diff
@@ -0,0 +1,354 @@
+            try:
+                connection.remove_role_from_instance_profile(InstanceProfileName=profile['InstanceProfileName'], RoleName=params['RoleName'])
+            except ClientError as e:
+                module.fail_json(msg=e.message, **camel_dict_to_snake_dict(e.response))
+
+        # Now remove any attached policies otherwise deletion fails
+        try:
+            for policy in get_attached_policy_list(connection, params['RoleName']):
+                connection.detach_role_policy(RoleName=params['RoleName'], PolicyArn=policy['PolicyArn'])
+        except (ClientError, ParamValidationError) as e:
+            module.fail_json(msg=e.message, **camel_dict_to_snake_dict(e.response))
+
+        try:
+            connection.delete_role(**params)
+        except ClientError as e:
+            module.fail_json(msg=e.message, **camel_dict_to_snake_dict(e.response))
+    else:
+        module.exit_json(changed=False)
+
+    module.exit_json(changed=True)
+
+
+def get_role(connection, name):
+
+    params = dict()
+    params['RoleName'] = name
+
+    try:
+        return connection.get_role(**params)['Role']
+    except ClientError as e:
+        if e.response['Error']['Code'] == 'NoSuchEntity':
+            return None
+        else:
+            module.fail_json(msg=e.message, **camel_dict_to_snake_dict(e.response))
+
+
+def get_attached_policy_list(connection, name):
+
+    try:
+        return connection.list_attached_role_policies(RoleName=name)['AttachedPolicies']
+    except ClientError as e:
+        if e.response['Error']['Code'] == 'NoSuchEntity':
+            return None
+        else:
+            module.fail_json(msg=e.message, **camel_dict_to_snake_dict(e.response))
+
+
+def main():
+
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(
+        dict(
+            name=dict(required=True, type='str'),
+            path=dict(default="/", required=False, type='str'),
+            assume_role_policy_document=dict(required=False, type='str'),
+            managed_policy=dict(default=[], required=False, type='list'),
+            state=dict(default=None, choices=['present', 'absent'], required=True)
+        )
+    )
+
+    module = AnsibleModule(argument_spec=argument_spec,
+                           required_if=[
+                               ('state', 'present', ['assume_role_policy_document'])
+                           ]
+                           )
+
+    if not HAS_BOTO3:
+        module.fail_json(msg='boto3 required for this module')
+
+    region, ec2_url, aws_connect_params = get_aws_connection_info(module, boto3=True)
+
+    connection = boto3_conn(module, conn_type='client', resource='iam', region=region, endpoint=ec2_url, **aws_connect_params)
+
+    state = module.params.get("state")
+
+    if state == 'present':
+        create_or_update_role(connection, module)
+    else:
+        destroy_role(connection, module)
+
+from ansible.module_utils.basic import *
```

Comment:
Is this a new requirement?  I've always written my modules like this.  Is there doc for likely necessary includes?